src/openlcb/SimpleStack.hxx

```diff
@@ -456,6 +457,60 @@ private:
     std::unique_ptr<PhysicalIf> create_if(const openlcb::NodeID node_id);
 };
 
+class SimpleTcpStackBase : public SimpleStackBase
+{
+public:
+    SimpleTcpStackBase(const openlcb::NodeID node_id);
+
+    /// @return the device representing the connection to the TCP hardware
+    /// link.
+    HubFlow *tcp_hub()
+    {
+        return &static_cast<TcpPhysicalIf *>(ifaceHolder_.get())->tcpHub0_;
+    }
+
+protected:
+    /// Helper function for start_stack et al.
+    void start_iface(bool restart) override;
+
+private:
+    class TcpPhysicalIf : public PhysicalIf
+    {
+    public:
+        TcpPhysicalIf(const openlcb::NodeID node_id, Service *service)
+            : tcpHub0_(service)
+            , ifTcp_(node_id, &tcpHub0_, config_local_nodes_count())
+            , datagramService_(&ifTcp_, config_num_datagram_registry_entries(),
+                  config_num_datagram_clients())
+        {
+        }
+
+        ~TcpPhysicalIf()
+        {
+        }
+
+        /// @return the OpenLCB interface object. Ownership is not transferred.
+        If *iface() override
+        {
+            return &ifTcp_;
+        }
+        /// @return the Datagram service bound to the interface. Ownership is
+        /// not transferred.
+        DatagramService *datagram_service() override
+        {
+            return &datagramService_;
+        }
+        HubFlow tcpHub0_;
+        IfTcp ifTcp_;
+        TcpDatagramService datagramService_;
```

Comment:
Should we get doxygen notes for these vars?

Reply:
Done.